---
title: "Metagenomics Analysis VEGA project"
subtitle: "GMM analysis"
author: "Sudarshan"
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: 
  html_document: 
    toc: yes
    toc_depth: 2
    toc_float: true
    code_folding: hide  
editor_options: 
  chunk_output_type: console
---

Scale mean of all sample abundance/sample abundance

# Background  

## Load pkgs  
```{r pkg, message=FALSE, warning=FALSE}

suppressPackageStartupMessages({
  library(vegan)
  library(data.table)
  library(ggpubr)
  library(ggplot2)
  library(microbiome)
  library(microbiomeutilities)
  library(reshape2)
  library(dplyr)
  library(patchwork)
  library(rstatix)
})
cols_grps <- c(
  Omnivore = "#ef476f",
  Pescatarian = "#f2cc8f",
  Vegan = "#4ecdc4",
  Vegetarian = "#1a936f"
)
source('codes/customFuns.R')

dir.create("results/metabolic/")
dir.create("results/metabolic/tab")
dir.create("results/metabolic/fig")
```

```{r}
meta_tab <- as.data.frame(fread("data/metagenome_vega_sample_selection.txt",
                                header = T, sep = "\t", 
                                stringsAsFactors = F))

meta_tab$MG_Sample_ID <- gsub("mg_Vega_", "VEGA", meta_tab$MG_Sample_ID)
rownames(meta_tab) <- meta_tab$MG_Sample_ID
meta_tab <- meta_tab[, -1]
colnames(meta_tab)[1] <- "unqiueID"
meta_tab$sampleID <- rownames(meta_tab)
dim(meta_tab)
head(meta_tab)
meta_tab <- meta_tab %>% 
  mutate(Group=ifelse(Group== "Meat consumer", "Omnivore", Group))
```

## Read GMMs  
```{r}
gmm.df <- readRDS("data/rds/gmm_tx.rds")
gmm.df$Taxon <- sub("s__", "", gmm.df$Taxon)
#head(gmm.df)
colnames(gmm.df) <- gsub(" ", "_", colnames(gmm.df))
gmm.df$Level_2 <- gmm.df$Level_3

gmm.df <- gmm.df %>% 
  mutate(Group=ifelse(Group== "Meat consumer", "Omnivore", Group))

```

```{r}
comps <- microbiomeutilities::make_pairs(gmm.df$Group)
```

### Amino acid degradation   

```{r fig.height=10, fig.width=14}
aad.df <- getFilteredGMM(gmm.df,
                         filter_level = "Level_1",
                         for_var = "amino acid degradation")

aad.ab.plot <- plotGMMAbundance(aad.df) + 
  labs(subtitle = "Amino acid degradation") + 
  theme(axis.text.x = element_blank(),
        axis.title.x = element_blank()) 
aad.sp.plot <- plotTopContributors(aad.df) 

p.aad <- aad.ab.plot / aad.sp.plot + plot_layout(widths = c(2,1))
p.aad
ggsave("results/metabolic/fig/amino_acid_species.pdf", h=6, w=6)

```


### Carbohydrate degradation  
```{r fig.height=10, fig.width=14}
crb.df <- getFilteredGMM(gmm.df,
                         filter_level = "Level_1",
                         for_var = "carbohydrate degradation")

carb.ab.plot <- plotGMMAbundance(crb.df) + 
  labs(subtitle = "Carbohydrate degradation") + 
  theme(axis.text.x = element_blank(),
        axis.title.x = element_blank())
carb.sp.plot <- plotTopContributors(crb.df) 

#getTopContributors(crb.df, groups_com = unique(crb.df$Group))

p.carb <- carb.ab.plot / carb.sp.plot + plot_layout(widths = c(2,1))
p.carb
ggsave("results/metabolic/fig/carbs_species.pdf", h=6, w=6)

```

### Glycoprotein degradation  
```{r fig.height=10, fig.width=14}
grpdeg.df <- getFilteredGMM(gmm.df,
                            filter_level = "Level_1",
                            for_var = "glycoprotein degradation")

grpdeg.ab.plot <- plotGMMAbundance(grpdeg.df) + 
  labs(subtitle = "Glycoprotein degradation") + 
  theme(axis.text.x = element_blank(),
        axis.title.x = element_blank())
grpdeg.sp.plot <- plotTopContributors(grpdeg.df) 

p.glyp <- grpdeg.ab.plot / grpdeg.sp.plot + plot_layout(heights = c(1,2))
p.glyp
ggsave("results/metabolic/fig/glycoprotein_species.pdf", h=6, w=6)

```


### Lipid degradation  
```{r fig.height=10, fig.width=14}
lipid.df <- getFilteredGMM(gmm.df,
                           filter_level = "Level_1",
                           for_var = "lipid degradation")

lipid.ab.plot <- plotGMMAbundance(lipid.df) + 
  labs(subtitle = "Lipid degradation") + 
  theme(axis.text.x = element_blank(),
        axis.title.x = element_blank())
lipid.sp.plot <- plotTopContributors(lipid.df)  

p.lip <- lipid.ab.plot / lipid.sp.plot + plot_layout(heights = c(1,2))
p.lip
ggsave("results/metabolic/fig/lipid_species.pdf", h=6, w=6)

```

```{r eval=FALSE}

(aad.ab.plot / aad.sp.plot) + (carb.ab.plot / carb.sp.plot) + plot_layout(widths = c(2,1,4,2)) 

(aad.ab.plot | carb.ab.plot) / (aad.sp.plot | carb.sp.plot)

(p.aad / p.carb) | (p.glyp) / (p.lip)
```


## SCFAs

```{r}
acetate.metabolism.df <- getFilteredGMM(gmm.df,
                                        filter_level = "Level_2",
                                        for_var = "acetate metabolism")

acetate.metabolismd.df.ab.plot <- plotGMMAbundance(acetate.metabolism.df) + 
  labs(subtitle = "acetate metabolism") + 
  theme(axis.text.x = element_blank(),
        axis.title.x = element_blank())
acetate.metabolism.sp.plot <- plotTopContributors(acetate.metabolism.df, top.bugs=6)

acet.gmm <- acetate.metabolismd.df.ab.plot/acetate.metabolism.sp.plot + plot_layout(heights = c(1,2))
acet.gmm
ggsave("results/metabolic/fig/acetate_species.pdf", h=6, w=6)

```

```{r}
lactate.metabolism.df <- getFilteredGMM(gmm.df,
                                        filter_level = "Level_2",
                                        for_var = "lactate metabolism")

lactate.metabolismd.df.ab.plot <- plotGMMAbundance(lactate.metabolism.df) + 
  labs(subtitle = "lactate metabolism") + 
  theme(axis.text.x = element_blank(),
        axis.title.x = element_blank())
lactate.metabolism.sp.plot <- plotTopContributors(lactate.metabolism.df, top.bugs=6)

lact.gmm <- lactate.metabolismd.df.ab.plot/lactate.metabolism.sp.plot + plot_layout(heights = c(1,2))
lact.gmm[[1]] + theme(legend.position = "top")
ggsave("results/metabolic/fig/legend.pdf", h=6, w=8)
ggsave("results/metabolic/fig/lactate_species.pdf", h=6, w=6)

```


```{r}
propionate.metabolism.df <- getFilteredGMM(gmm.df,
                                           filter_level = "Level_2",
                                           for_var = "propionate metabolism")

propionate.metabolismd.df.ab.plot <- plotGMMAbundance(propionate.metabolism.df) + 
  labs(subtitle = "propionate metabolism") + 
  theme(axis.text.x = element_blank(),
        axis.title.x = element_blank())
propionate.metabolism.sp.plot <- plotTopContributors(propionate.metabolism.df, top.bugs=6)

prop.gmm <- propionate.metabolismd.df.ab.plot/propionate.metabolism.sp.plot + plot_layout(heights = c(1,2))
prop.gmm
ggsave("results/metabolic/fig/propionate_species.pdf", h=6, w=6)

```


```{r}
butyrate.metabolism.df <- getFilteredGMM(gmm.df,
                                         filter_level = "Level_2",
                                         for_var = "butyrate metabolism")

butyrate.metabolismd.df.ab.plot <- plotGMMAbundance(butyrate.metabolism.df) + 
  labs(subtitle = "butyrate metabolism") + 
  theme(axis.text.x = element_blank(),
        axis.title.x = element_blank())
butyrate.metabolism.sp.plot <- plotTopContributors(butyrate.metabolism.df, top.bugs=6)

but.gmm <- butyrate.metabolismd.df.ab.plot/butyrate.metabolism.sp.plot + plot_layout(heights = c(1,2))
but.gmm
ggsave("results/metabolic/fig/butyrate_species.pdf", h=6, w=6)

```

# Reviewer  

```{r}
(aad.ab.plot + carb.ab.plot) / (grpdeg.ab.plot + lipid.ab.plot) + plot_annotation(tag_levels = "a")
ggsave("results/metabolic/fig/GMM_level_1.pdf", h=5, w=9)

(acetate.metabolismd.df.ab.plot + lactate.metabolismd.df.ab.plot) / 
  (propionate.metabolismd.df.ab.plot + butyrate.metabolismd.df.ab.plot) + plot_annotation(tag_levels = "a")
ggsave("results/metabolic/fig/GMM_level_2.pdf", h=5, w=9)

p.l1 <- (aad.ab.plot + carb.ab.plot) / (grpdeg.ab.plot + lipid.ab.plot) + plot_annotation(tag_levels = "a")
p.l2 <- (acetate.metabolismd.df.ab.plot + lactate.metabolismd.df.ab.plot) / 
  (propionate.metabolismd.df.ab.plot + butyrate.metabolismd.df.ab.plot) + plot_annotation(tag_levels = "a")


ggsave("results/metabolic/fig/GMM_level_1n2.pdf", h=9, w=12)
```


## Significant contributors  
```{r}
plotSigTaxa <- function(df,
                        abund_cutoff = 0,
                        prevalence.cutoff = 5/100,
                        fdr.method = "BH"){
 
  wide.df <- df %>% 
    pivot_wider(names_from = variable, id_cols = Taxon, values_from = abundance) %>% 
    as.data.frame() %>% 
    column_to_rownames("Taxon")
  keep <- core_members(wide.df, abund_cutoff, prevalence.cutoff) 
  df <- df %>% filter(Taxon %in% keep)
  
  sdh <- df %>%
    left_join(meta_tab, by= c("variable" ="sampleID")) %>% 
    group_by(Group.x, variable) %>% 
    mutate(frequency = abundance / sum(abundance)) %>% 
    mutate(frequency = frequency*100) %>% 
    ungroup()
  
  sdh.test <- sdh %>% 
    group_by(Taxon) %>% 
    kruskal_test(frequency ~ Group.x) %>%
    adjust_pvalue(method = fdr.method) %>%
    add_significance("p.adj")
  
  # sdh.test |> 
  #   filter(p.adj <= 0.05)
  
  dun.test <- sdh %>% 
    group_by(Taxon) %>% 
    dunn_test(frequency ~ Group.x) %>%
    adjust_pvalue(method = fdr.method) %>%
    add_significance("p.adj")  
  
  sig.dat <- dun.test |> 
    mutate(comparison = paste0(group1, " vs ", group2)) |> 
    mutate(diff = ifelse(p.adj < 0.05, "+", "-"))
  
  p1 <- sig.dat |> 
    filter(p.adj < 0.05) |> 
    filter(Taxon !="unclassified") |>
    mutate(SigDif = "Different") |> 
    ggplot(aes(y= as.factor(Taxon), x= comparison)) +
    geom_tile(aes(fill=diff), color="grey90", show.legend = FALSE)+
    scale_fill_manual(values = "brown3") +
    theme_vega()+
    rotate_x_text(angle = 45)+
    facet_grid(~SigDif, scales = "free_x")+
    theme(axis.title = element_blank(),
          axis.text.y = element_text(face = "italic")) 
  
  sig.tax <- dun.test |> 
    filter(p.adj < 0.05)
  sdh.sig.tax <- sdh |> 
    filter(Taxon %in% unique(sig.tax$Taxon))
  #sdh.sig.tax
  
  p2 <- sdh.sig.tax |> 
    filter(Taxon !="unclassified") |>
    ggplot(aes(x = variable, y = as.factor(Taxon))) +
    geom_tile(aes(fill= log10(frequency+1))) +
    scale_fill_gradientn("Relative \ncontribution (log10)",
                         colours = c("#f8fafd", "#1d3557"),
                         #trans = "log10",
                         na.value = "white") +
    theme_vega() +
    rotate_x_text(angle = 45) +
    facet_grid(~Group.x, scales = "free_x")+
    theme(axis.title = element_blank(),
          panel.grid.major =  element_blank(),
          strip.background = element_rect(color="grey70"),
          axis.text.y = element_blank(),
          axis.text.x = element_blank()) 
  
  return((p1 + p2) + plot_layout(widths = c(0.3,1)))
}
```

## Level 1  
```{r}
p.aa <- plotSigTaxa(aad.df) + labs(subtitle = "amino acid degradation")
p.cb <- plotSigTaxa(crb.df) + labs(subtitle = "carbohydrate degradation")
p.gp <- plotSigTaxa(grpdeg.df)+ labs(subtitle = "glycoprotein degradation")
p.lp <- plotSigTaxa(lipid.df)+ labs(subtitle = "lipid degradation")
(p.aa / p.cb /p.gp) + plot_layout(heights = c(0.8,0.8,0.4)) + plot_annotation(tag_levels = "a")
ggsave("results/metabolic/fig/GMM_Taxa_level_1.pdf", h=8, w=12)
```

## Level 2  
```{r}
p.ace <- plotSigTaxa(acetate.metabolism.df)+ labs(subtitle = "acetate metabolism")
p.lac <- plotSigTaxa(lactate.metabolism.df)+ labs(subtitle = "lactate metabolism")
p.pro <- plotSigTaxa(propionate.metabolism.df)+ labs(subtitle = "propionate metabolism")
p.but <- plotSigTaxa(butyrate.metabolism.df)+ labs(subtitle = "butyrate metabolism")

(p.ace / p.lac /p.pro) + plot_layout(heights = c(0.5,1,0.8)) + plot_annotation(tag_levels = "a")
ggsave("results/metabolic/fig/GMM_Taxa_level_2.pdf", h=7, w=11)

```

```{r}
sessionInfo()
```







